### DISCUSSION TITLE
Migrate from "master" to "main" but keep "master"

### DISCUSSION THREAD

**Discussion by buhtz** [community/community#34325] [Other Feature Feedback, Questions, & Ideas] [answered]
There are a lot of HowTo's and tutorials around explaining how to "rename" the primary branch from `master` to `main`. No big deal.

But I want to keep `master` for historical reasons and migrate to `main` as the primary/base branch.

How can I archive this? The `main` branch should have the whole history of the `master` branch.

There are a lot of extern projects (e.g. distro maintainers) referencing my `master` branch. I don't know them all.
I would create a last commit on that `master` branch making my application raise an Exception informing about the situation and pointing to the correct `main` branch.

**Comment by dgholz** [comment, marked as answer]
Multiple branches can point to the same commit, no problem. After renaming the old `master` branch, just create a new one pointing to `main`. From the command line:

    git push origin main:master # pushing local main branch to master on remote

or through the API:

    main_commit_id="$( gh api repos/{owner}/{repo}/git/ref/heads/master -q .object.sha )"
    gh api repos/{owner}/{repo}/git/refs -f ref=refs/heads/master -f sha="$main_commit"

Then, just work on `main` and ignore `master`. `master` will stay pointing to that old commit forever.

**Reply by buhtz** [reply]
Thanks I will test that.
